base/logging.jl

```diff
@@ -290,7 +290,13 @@ function logmsg_code(_module, file, line, level, message, exs...)
     end
     # Note that it may be necessary to set `id` and `group` manually during bootstrap
     id = something(id, :(log_record_id(_module, level, $exs)))
-    group = something(group, :(Symbol(splitext(basename(something($file, "")))[1])))
+    if group == nothing
+        group = if isdefined(Base, :basename) && isa(file, String)
+            QuoteNode(splitext(basename(something(file, "")))[1])
+        else
+            :(Symbol(splitext(basename(something($file, "")))[1]))
```

Comment:
maybe memoize this in addition or instead?
```julia
Ref = Ref{Symbol}()
:(isassigned($Ref) ? $Ref[] : ($Ref[] = ...))
```

Reply:
Good idea, added.